### ISSUE TITLE
README examples are broken

### ISSUE THREAD

**Issue by martinandersson** [renatoathaydes/setup-java-sdkman#3]
The "basic" version is not valid, missing the "on:", "jobs:" and "build:" elements.

The "matrix testing" is missing "on:", and is using a non-existent Java version ("8.0.265-zulu"), and is referencing a file that does not exist ("HelloWorldApp").

The final _working_ example of matrix testing (incorporating fix for #2) is:

    on: [push, pull_request]
    jobs:
      build:
        runs-on: ubuntu-16.04
        strategy:
          matrix:
            java: [ "8.0.282-zulu", "11.0.2-open" ]
        name: Java ${{ matrix.java }} sample
        steps:
          - uses: actions/checkout@v2
          - name: Setup java
            uses: renatoathaydes/setup-java-sdkman@v2.1.0
            with:
              java-version: ${{ matrix.java }}
          - run: java -version

**Comment by renatoathaydes** [comment]
It worked before, but GH Actions kept changing.  That zulu version existed before, but SDKMAN keeps deleting old versions unfortunately.

**Comment by renatoathaydes** [comment]
Oh you mean the first example? Of course that won't work, as I only documented the parts specific to this action. Sorry if that was unclear.

**Comment by TarqusMiltonium** [comment]
I noticed the same thing -- but every other SDKMan Action seems to suffer the same fate of older versions disappearing.

I wonder if we could improve this Action by dynamically parsing the output of sdkman and finding the latest version? This could perhaps be done with a set of regexes, or possibly construct one?

Eg:
```
java: [ "8-amzn", "11-open"]
```

-->

```
8_amzn=$(sdk list java | grep amzn | grep 8 | head -n 1 | grep -o '8[0-9\.]\+-amzn')
11_open=$(sdk list java | grep open | grep 11 | head -n 1 | grep -o '11[0-9\.]\+-open')
```
?

**Comment by renatoathaydes** [comment]
I agree that due to how SDKMAN works, this plugin is only useful if we are able to change which JDK version to use based on what's currently available. SDKMAN seems to be very aggressive in removing previous minor versions (implying they want you to be on the latest version, always, as Oracle seems to prefer with the new release cadence).

So I think we should have something like this:

```
jdk_major_version = 16
vendor = azul
```

And then we ask SDKMAN what's the latest JDK 16 version from Azul (as an example).

**Comment by TarqusMiltonium** [comment]
Yep, I love it, I had a similar idea after I posted my last comment.

**Comment by TarqusMiltonium** [comment]
Hey, I was just going to do this, when I read the [relevant doco](https://docs.github.com/en/actions/learn-github-actions/finding-and-customizing-actions#using-release-management-for-your-custom-actions), specifically:

>  # Using branches
> 
> Specifying a target branch for the action means it will always run the version currently on that branch.

Perhaps it would be worth adding a `v2` or `v2.x` branch to this repo instead of updating the doco? This way the action will improve as you iterate it (and should breaking changes be required this can result in a new branch).  If the branch maintenance gets weird later, you can simply replace the branch with a tag of the final commit in that branch, and everyone's dependent Actions yaml references will continue to work the same (see above doco for the identical nature of yaml references for tag vs branch).  Using a branch will allow you to roll out any changes (eg better non-interactive running) to all users, whilst keeping doco requirements to a minimum.

It should be noted that as a branch, this can obviously induce breaking behaviour, as GitHub itself documents:
> This approach can create problems if an update to the branch includes breaking changes
However this can be worked around by consumers by using version tags and/or commits IDs (as already at the current time).

Thoughts?

**Comment by renatoathaydes** [comment]
@TarqusMiltonium that's great, I can introduce version branches that only get minor bug fixes introduced so that people can use them to keep in sync.
